Source/WebCore/rendering/style/StyleSVGResourceImage.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "CachedImageClient.h"
+#include "CachedResourceHandle.h"
+#include "StyleGeneratedImage.h"
+
+namespace WebCore {
+
+class StyleSVGResourceImage final : public StyleGeneratedImage {
+public:
+    static Ref<StyleSVGResourceImage> create(Ref<CSSImageValue> cssValue, const AtomString& fragmentIdentifier)
+    {
+        return adoptRef(*new StyleSVGResourceImage(WTFMove(cssValue), fragmentIdentifier));
+    }
+    virtual ~StyleSVGResourceImage();
+
+    bool operator==(const StyleImage& other) const final;
+    bool equals(const StyleSVGResourceImage&) const;
+    bool equalInputImages(const StyleSVGResourceImage&) const;
+
+    static constexpr bool isFixedSize = true;
```

Comment:
Is it fixed size though?

Reply:
This class is removed. And SVG maskers should not be fixed size.